### ISSUE TITLE
JSTL - Instructor Home: remove use of Logic.getCourseIdSectionNamesMap from Page Data

### ISSUE THREAD

**Issue by kanghj** [TEAMMATES/repo#3794]
Essentially #3735 done on the JSTL branch. This needs to be done before #3743 can be merged into JSTL-Dev, then JSTL-Dev into master. 

see #3745
